app/pkg/protocol/channel.go

```diff
@@ -23,20 +23,20 @@ var (
 
 // Channel represents information about current state, channel info.
 type Channel struct {
-	initProposal InitProposal
+	initProposal *InitProposal
 	lastState    *state.State
-	c            *channel.Channel
+	c            chl.Channel
 }
 
 // InitChannel opens channel with participant who was requested opening a channel.
-func InitChannel(initProposal InitProposal, participantIndex uint) (*Channel, error) {
-	c, err := channel.New(*initProposal.State, participantIndex)
+func InitChannel(initProposal *InitProposal, participantIndex uint) (*Channel, error) {
+	c, err := chl.New(*initProposal.State, participantIndex)
```

Comment:
here as you see they return the copy, that's why I don't really see a reason to store a pointer in our side (its from go-nitro library)

you can see here that with New function they return copy and not a pointer 😅